__tests__/atlas.test.ts

```diff
@@ -4,12 +4,40 @@ import { tmpdir } from 'os'
 import { getExecOutput } from '@actions/exec'
 import * as path from 'path'
 import { SimpleGit, simpleGit } from 'simple-git'
-import { copyFile, mkdir, mkdtemp, readdir, rm, stat } from 'fs/promises'
-import { AtlasResult, ExitCodes, installAtlas } from '../src/atlas'
+import {
+  copyFile,
+  mkdir,
+  mkdtemp,
+  readdir,
+  readFile,
+  rm,
+  stat
+} from 'fs/promises'
 import * as core from '@actions/core'
 import * as github from '@actions/github'
 import nock from 'nock'
 import * as http from '@actions/http-client'
+import { AtlasResult, ExitCodes, installAtlas } from '../src/atlas'
+import {
+  ARCHITECTURE,
+  BASE_ADDRESS,
+  getDownloadURL,
+  LATEST_RELEASE,
+  S3_FOLDER
+} from '../src/cloud'
+
+jest.mock('../src/cloud', () => {
+  const actual = jest.requireActual('../src/cloud')
+  return {
+    ...actual,
+    getDownloadURL: jest.fn(
+      (version: string) =>
+        new URL(
+          `${BASE_ADDRESS}/${S3_FOLDER}/atlas-${ARCHITECTURE}-${version}?test=1`
```

Comment:
just saying, you could have a helper function that adds the test=1 query param, since it's used multiple times

Reply:
It means that I will have to check if I'm inside a test inside the actual code, really ugly.